### PR TITLE
retry curl for up to 30 seconds

### PR DESCRIPTION
**Pull request by bacongobbler**

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -31,7 +33,7 @@ if [ -f $BUILD_DIR/.crystal-version ]; then
   CRYSTAL_VERSION=`cat $BUILD_DIR/.crystal-version | tr -d '\012'`
   CRYSTAL_VERSION_REASON='due to .crystal-version file'
 else
-  CRYSTAL_VERSION=`curl -s https://api.github.com/repos/crystal-lang/crystal/releases/latest | json_value tag_name 1 | tr -d '\012'`
+  CRYSTAL_VERSION=`${CURL} -s https://api.github.com/repos/crystal-lang/crystal/releases/latest | json_value tag_name 1 | tr -d '\012'`
```
Comment (bcardiff) [inline bin/compile]: -s flag is duplicated due to CURL

File: bin/compile [hunk 2]
```diff
@@ -42,7 +44,7 @@ unset GIT_DIR
 # Install Crystal
 echo "-----> Installing Crystal ($CRYSTAL_VERSION $CRYSTAL_VERSION_REASON)"
 mkdir -p $CRYSTAL_DIR
-curl -sL $CRYSTAL_URL | tar xz -C $CRYSTAL_DIR --strip-component=1
+${CURL} -sL $CRYSTAL_URL | tar xz -C $CRYSTAL_DIR --strip-component=1
```
Comment (bcardiff) [inline bin/compile]: -sL flag is duplicated due to CURL

**Review by bcardiff — CHANGES_REQUESTED** [review]
Some flags are duplicated during execution

**Comment by bacongobbler** [comment]
fixed

**Comment by bcardiff** [comment]
Thanks

**Pull request merged**